### ISSUE TITLE
Newest Flow Log are not showing

### ISSUE THREAD

**Issue by FutureBuddha** [directus/directus#23848]
### Describe the Bug

When executing a flow in Directus, the latest logs are not always displayed in real-time. In some cases, they appear after a delay, and in others, it can take several hours for the logs to show up. This inconsistent behavior makes it difficult to monitor flow executions effectively.

https://github.com/user-attachments/assets/d85f8306-d189-49e6-8664-868a48378a8b

### To Reproduce

1. Create a flow that is manually triggered.
2. Execute the flow.
3. Monitor the logs section for newly created log entries.
4. Repeat the process multiple times.

**Expected Behavior:**

Logs should be displayed immediately after the flow execution, allowing users to view real-time feedback for each trigger.

**Actual Behavior:**

- Sometimes the logs appear as expected.
- Other times, there is a significant delay (up to several hours) before the logs are visible.
- This issue appears to occur more frequently as the number of logs increases, suggesting it could be related to the volume of log data.

**Additional Information:**

- The issue seems intermittent; at the beginning, logs are displayed correctly, but the problem becomes more pronounced after repeated flow executions.
- The system may be experiencing delays due to a high number of logs being generated.

### Directus Version

v11.1.1

### Hosting Strategy

Self-Hosted (Docker Image)

**Comment by br41nslug** [comment]
Sounds like this could potentially be a caching issue, how have you configured caching for this setup if at all?

**Comment by FutureBuddha** [comment]
I am using a self-hosted setup with multiple instances, and the following cache settings are in place:

CACHE_TTL=7 days
CACHE_STORE=redis
CACHE_ENABLED=true
CACHE_AUTO_PURGE=true

**Comment by br41nslug** [comment]
You mention multiple instances, horizontally scaled i am assuming, have you configured the `SYNCHRONIZATION_STORE` to be redis instead of memory? This allows the individual instances to synchronize directly through redis https://docs.directus.io/self-hosted/config-options.html#synchronization

**Comment by arinrare** [comment]
This is happening for me too. I can get the logs to show if i delete the Docker container and recreate it

**Comment by rijkvanzanten** [comment]
@FutureBuddha Do you have a CDN sitting in between Directus and your computer? 🤔 

We haven't been able to reproduce this yet, so bear with

**Comment by FutureBuddha** [comment]
No, deploy i via docker on fly.io

**Comment by rijkvanzanten** [comment]
@FutureBuddha What database version and provider are you using? Sounds like something in the stack is caching somewhere, but I'm having trouble reproducing it without recreating your exact hosting environment

**Comment by FutureBuddha** [comment]
PostgreSQL 16.3 on x86_64-pc-linux-gnu, compiled by gcc (Debian 10.2.1-6) 10.2.1 20210110, 64-bit , postgis_version 1 3.3 USE_GEOS=1 USE_PROJ=1 USE_STATS=1, from https://neon.tech/


do you need somthing else?

**Comment by AufarZakiev** [comment]
Hi, facing same issue.
Disabling cache has no effect. I use self-hosted Docker.

When logs are shown, there is a log from Directus container with 304 code:
```
[23:27:00] GET /revisions?filter[collection][_eq]=directus_flows&filter[item][_eq]=0a128f0f-4ef9-457b-8ad4-f6a8549fea43&filter[version][_null]=true&filter[activity][action][_eq]=create&sort=-id&limit=1&fields[]=id&fields[]=data&fields[]=delta&fields[]=collection&fields[]=item&fields[]=activity.action&fields[]=activity.timestamp&fields[]=activity.user.id&fields[]=activity.user.email&fields[]=activity.user.first_name&fields[]=activity.user.last_name&fields[]=activity.ip&fields[]=activity.user_agent&fields[]=activity.origin 304 14ms
```

As far as I understand, that exact request is used for logs. Is `filter[version][_null]=true` part OK ? This `_null` seems suspicious.

**Comment by rijkvanzanten** [comment]
Yeah that's OK, [the `_null` operator](https://directus.io/docs/guides/connect/filter-rules#available-operators) is to check whether a field is `null` or not 👍🏻

**Comment by AufarZakiev** [comment]
Ok. Don't know what happened but it started to work after the one of changes. Btw, recreating the same flow once again didn't work.

I will try to find reproducible flow for you. Seems like it is somehow related to request URL operations. Maybe log is not created till the end of the flow and the flow never ends if URL does not response...

**Comment by rijkvanzanten** [comment]
Ohh interesting! Thanks for the heads up, that does sound like something worth exploring indeed. A flow unintentionally never completing would indeed cause logs to be missing 🤔

**Comment by AufarZakiev** [comment]
Well, I tried to get a reproducible example but gave up.
It would be better to just make a donation :) Thanks for a great tool!
